auth/service_test.go

```diff
@@ -379,3 +442,185 @@ func TestDBAuthService_Authorize(t *testing.T) {
+	if timeDiff > time.Second || timeDiff < -1*time.Second {
+		t.Errorf("expected to receive issued date close to %s, got %s (diff %s)", credential.IssuedDate, gotCredential.IssuedDate, timeDiff)
+	}
+	// TODO(ariels): add more credentials (and test)
+}
+
+func TestDBAuthService_ListGroups(t *testing.T) {
+	cases := []struct {
+		name       string
+		groupNames []string
+	}{
+		{
+			name:       "no_groups",
+			groupNames: []string{},
+		}, {
+			name:       "single_group",
+			groupNames: []string{"fooers"},
+		}, {
+			name:       "many_groups",
+			groupNames: []string{"fooers", "barriers", "bazaars", "quuxers", "pling-plongers"},
+		},
+	}
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			s := setupService(t)
+			for _, groupName := range testCase.groupNames {
+				if err := s.CreateGroup(&model.Group{DisplayName: groupName}); err != nil {
+					t.Fatalf("CreateGroup(%s): %s", groupName, err)
+				}
+			}
+			gotGroupNames := make([]string, 0, len(testCase.groupNames))
+			groups, _, err := s.ListGroups(&model.PaginationParams{Amount: -1})
+			if err != nil {
+				t.Errorf("ListGroups: %s", err)
+			}
+			for _, group := range groups {
+				gotGroupNames = append(gotGroupNames, group.DisplayName)
+			}
+			sort.Strings(testCase.groupNames)
+			sort.Strings(gotGroupNames)
+			if diffs := deep.Equal(testCase.groupNames, gotGroupNames); diffs != nil {
+				t.Errorf("got different groups than expected: %s", diffs)
+			}
+		})
+	}
+}
+
+func TestDbAuthService_GetUser(t *testing.T) {
+	s := setupService(t)
+	const userName = "foo"
+	// Time should *not* have nanoseconds - otherwise we are comparing accuracy of golang
+	// and Postgres time storage.
+	time := time.Date(2222, 2, 22, 22, 22, 22, 0, time.UTC)
+	if err := s.CreateUser(&model.User{DisplayName: userName, ID: -22, CreatedAt: time}); err != nil {
+		t.Fatalf("CreateUser(%s): %s", userName, err)
+	}
+	user, err := s.GetUser(userName)
+	if err != nil {
+		t.Fatalf("GetUser(%s): %s", userName, err)
+	}
+	if user.DisplayName != userName {
+		t.Errorf("GetUser(%s) returned user %+v with a different name", userName, user)
+	}
+	if user.CreatedAt.Sub(time) != 0 {
+		t.Errorf("expected user CreatedAt %s, got %+v", time, user.CreatedAt)
+	}
+	if user.ID == -22 {
+		t.Errorf("expected CreateUser ID:-22 to be dropped on server, got user %+v", user)
+	}
+}
+
+func TestDbAuthService_GetUserById(t *testing.T) {
+	s := setupService(t)
+	const userName = "foo"
+	// Time should *not* have nanoseconds - otherwise we are comparing accuracy of golang
+	// and Postgres time storage.
+	time := time.Date(2222, 2, 22, 22, 22, 22, 0, time.UTC)
+	if err := s.CreateUser(&model.User{DisplayName: userName, ID: -22, CreatedAt: time}); err != nil {
+		t.Fatalf("CreateUser(%s): %s", userName, err)
+	}
+	user, err := s.GetUser(userName)
+	if err != nil {
+		t.Fatalf("GetUser(%s): %s", userName, err)
+	}
+	userById, err := s.GetUserByID(user.ID)
```

Comment:
userById is a bit misleading as a name, sounds like a map

Reply:
Changed to `gotUser`.